utils.py

```diff
@@ -0,0 +1,112 @@
+from syslog import LOG_DEBUG
+
+import iso8601
+from pyparsing import Word, Suppress, nums, Optional, Regex, pyparsing_common, alphanums
+
+
+def parse_log_lines(body):
+    parser = Parser()
+    return [l for l in parser.parse_rfc6587(body) if filter_log_lines(l)]
+
+
+def filter_log_lines(msg):
+    """Return true if we should send to kinesis."""
+    sev = msg["severity"]  # e.g. LOG_DEBUG
+    body = msg["message"]
+
+    if sev >= LOG_DEBUG:
+        return False
+
+    if body.startswith("DEBUG   "):
+        return False
+
+    return True
+
+
+class Parser(object):
+    def __init__(self):
+        ints = Word(nums)
+
+        # priority
+        priority = Suppress("<") + ints + Suppress(">")
+
+        # version
+        version = ints
+
+        # timestamp
+        timestamp = pyparsing_common.iso8601_datetime
+
+        # hostname
+        hostname = Word(alphanums + "_" + "-" + ".")
+
+        # source
+        source = Word(alphanums + "_" + "-" + ".")
+
+        # appname
+        appname = (
+            Word(alphanums + "(" + ")" + "/" + "-" + "_" + ".")
+            + Optional(Suppress("[") + ints + Suppress("]"))
+            + Suppress("-")
+        )
+
+        # message
+        message = Regex(".*")
+
+        # pattern build
+        self.__pattern = (
+            priority + version + timestamp + hostname + source + appname + message
+        )
+
+    def get_chunk(self, payload):
+        msg_len, syslog_msg_payload = payload.split(" ", maxsplit=1)
+        msg_len = int(msg_len)
+
+        # FIXME
+        # There is a problem with directly using the MSGLEN in the current configuration. What seems to be
+        # happening is that when a syslog line is sent from Heroku, a unicode character is counted as having
+        # a length of 2. But, when the loglines are being processed within this python 3.6 code, all unicode
+        # characters have a length of 1.
+        #
+        # What exists below is a _bad hack_, but when working on this I was having trouble figuring out how to
+        # accomplish what I want to accomplish, which seems to be converting unicode into its control characters
+        # or similar for the purpose of parsing out the message.
+        num_of_unicode_chars = len(
+            [c for c in syslog_msg_payload[0:msg_len] if ord(c) > 128]
+        )
+        msg_len -= num_of_unicode_chars
```

Comment:
I would love some assistance on what is the best way to accomplish the above.

#### As well, here is an example:
```
In [16]: logline = '155 <190>1 2018-08-03T22:38:42.975461+00:00 host app web.1 - [GIN] 2018/08/03 - 22:
    ...: 38:42 | 200 |     170.221µs |  256.0.0.10 | GET      /static/lang-logo.png\n146 <190>1 2018-08
    ...: -03T22:38:43.087424+00:00 host app web.1 - [GIN] 2018/08/03 - 22:38:43 | 404 |       1.973µs |
    ...:   256.0.0.10 | GET      /favicon.ico\n'

In [17]: len("<190>1 2018-08-03T22:38:42.975461+00:00 host app web.1 - [GIN] 2018/08/03 - 22:38:42 | 20
    ...: 0 |     170.221µs |  256.0.0.10 | GET      /static/lang-logo.png\n")
Out[17]: 154

In [18]: logline[4:]
Out[18]: '<190>1 2018-08-03T22:38:42.975461+00:00 host app web.1 - [GIN] 2018/08/03 - 22:38:42 | 200 |     170.221µs |  256.0.0.10 | GET      /static/lang-logo.png\n146 <190>1 2018-08-03T22:38:43.087424+00:00 host app web.1 - [GIN] 2018/08/03 - 22:38:43 | 404 |       1.973µs |  256.0.0.10 | GET      /favicon.ico\n'

# The last character is the first digit in the next messages MSGLEN
In [19]: logline[4:][0:155]
Out[19]: '<190>1 2018-08-03T22:38:42.975461+00:00 host app web.1 - [GIN] 2018/08/03 - 22:38:42 | 200 |     170.221µs |  256.0.0.10 | GET      /static/lang-logo.png\n1'
```

Reply:
Is the MSGLEN a byte len vs str len thing? Can you used the byte len?

```
>>> len(bytes('¯\_(ツ)_/¯', 'utf8'))
13
>>> len('¯\_(ツ)_/¯')
9
```